UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs

```diff
@@ -450,46 +453,126 @@ private void RefreshAll()
 		public void RollRandomCharacter()
 		{
 			// Randomise gender
-			var changeGender = (UnityEngine.Random.Range(0, 2) == 0);
-			if (changeGender)
+			Type gender = typeof(BodyType);
+			Array genders = gender.GetEnumValues();
+			int index = UnityEngine.Random.Range(0,3);
+			currentCharacter.BodyType = (BodyType)genders.GetValue(index);
+
+			//Randomises player name and age.
+			switch (currentCharacter.BodyType)
 			{
-				//OnGenderChange();
+				case BodyType.Male:
+					currentCharacter.Name = StringManager.GetRandomMaleName();
+					break;
+				case BodyType.Female:
+					currentCharacter.Name = StringManager.GetRandomFemaleName();
+					break;
+				default:
+					currentCharacter.Name = StringManager.GetRandomName(Gender.NonBinary);  //probably should get gender neutral names?
+					break;																	//for now it will pick from both the male and female name pools
 			}
+			currentCharacter.Age = UnityEngine.Random.Range(19, 78);
+
+			//Randomises player accents. (Italian, Scottish, etc)
+			randomizeAccent();
+
 
-			// Select a random value from each dropdown
-			// hairDropdown.value = UnityEngine.Random.Range(0, hairDropdown.options.Count - 1);
-			// facialHairDropdown.value = UnityEngine.Random.Range(0, facialHairDropdown.options.Count - 1);
-			// underwearDropdown.value = UnityEngine.Random.Range(0, underwearDropdown.options.Count - 1);
-			// socksDropdown.value = UnityEngine.Random.Range(0, socksDropdown.options.Count - 1);
-				switch (currentCharacter.BodyType)
+			//Randomises character skin tones.
+			randomizeSkinTones();
+
+			//Randomises character clothes, cat ears, moth wings, etc.
+			randomizeEverything();
+
+			//Refresh the player character's sheet so they can see their new changes.
+			RefreshAll();
+		}
+
+		private void randomizeEverything()
+		{
+			int nudeChance = UnityEngine.Random.Range(0, 100);
+			foreach(var dropdown in GetComponentsInChildren<BodyPartDropDownReplaceOrgan>())
+			{
+				dropdown.RandomizeDropdownValue();
+			}
+			//Hair, tail, etc radomization.
+			foreach (var color in GetComponentsInChildren<BodyPartSpriteAndColour>())
+			{
+				if(currentCharacter.BodyType == BodyType.Female && color.RelatedBodyPart.name == "Facial Hair")
```

Comment:
humm this is a good point since this logic can't be covered under BodyPartSpriteAndColour maybe you could make a new class that inherits from BodyPartSpriteAndColour that handles body type conditional sprites?

Reply:
we probably don't need an entire new class for it, will introduce conditionals once I come back home.

Reply:
yeah is fine for now